packages/core/content-manager/server/src/controllers/relations.ts

```diff
@@ -45,117 +44,199 @@ const sanitizeMainField = (model: any, mainField: any, userAbility: any) => {
   return mainField;
 };
 
+const isNumeric = (value: any): value is number => {
+  const parsed = parseInt(value, 10);
+  return !Number.isNaN(parsed);
+};
+
+/**
+ *
+ */
+const mapResults = (results: Array<any>) => {
+  return results.map((result: any) => {
+    if (result.documentId !== undefined) {
+      result.id = result.documentId;
+      delete result.documentId;
+    }
+
+    return result;
+  });
+};
+
 export default {
-  async findAvailable(ctx: any) {
+  async extractAndValidateRequestInfo(
+    ctx: any,
+    status?: 'draft' | 'published',
+    id?: string | number,
+    locale?: string
+  ) {
     const { userAbility } = ctx.state;
     const { model, targetField } = ctx.params;
 
-    await validateFindAvailable(ctx.request.query);
-
-    // idsToOmit: used to exclude relations that the front already added but that were not saved yet
-    // idsToInclude: used to include relations that the front removed but not saved yes
-    const { entityId, idsToOmit, idsToInclude, _q, ...query } = ctx.request.query;
-
-    const modelSchema = strapi.getModel(model);
-    if (!modelSchema) {
-      return ctx.badRequest("The model doesn't exist");
+    const sourceSchema = strapi.getModel(model);
+    if (!sourceSchema) {
+      throw new errors.ValidationError(`The model ${model} doesn't exist`);
     }
 
-    const attribute: any = modelSchema.attributes[targetField];
+    const attribute: any = sourceSchema.attributes[targetField];
     if (!attribute || attribute.type !== 'relation') {
-      return ctx.badRequest("This relational field doesn't exist");
+      throw new errors.ValidationError(
+        `The relational field ${targetField} doesn't exist on ${model}`
+      );
     }
 
-    const isComponent = modelSchema.modelType === 'component';
+    const isSourceComponent = sourceSchema.modelType === 'component';
 
-    if (!isComponent) {
-      const permissionChecker = getService('permission-checker').create({
-        userAbility,
-        model,
-      });
+    const where: Record<string, any> = {};
+    if (!isSourceComponent) {
+      where.locale = locale;
+      where.publishedAt = status === 'published' ? { $ne: null } : null;
+    }
 
-      if (permissionChecker.cannot.read(null, targetField)) {
-        return ctx.forbidden();
+    let currentEntity = { id: null };
+    if (id) {
+      // The Id we receive can be a documentId or a numerical entity id
+      if (isNumeric(id)) {
+        // TODO is there a better way to distinguish between the two?
+        where.id = id;
+      } else {
+        where.documentId = id;
       }
```

Comment:
entry id being numeric might change in the future, I would try to avoid this as much as possible , we do something similar when parsing relations, but we do plan to change that too

Reply:
also, will it only be numeric when asking for a component relation?